ocean_lib/web3_internal/contract_base.py

```diff
@@ -48,7 +48,8 @@ def __init__(self, web3: Web3, address: Optional[str]) -> None:
         self.web3 = web3
         self.contract = load_contract(self.web3, self.name, address)
         assert not address or (
-            self.contract.address == address and self.address == address
+            self.contract.address.lower() == address.lower()
```

Comment:
no extra lowering. If the addresses are both checksummed then it is correct. In this case, I would also checksum the address first thing and alert the user if the address can not be converted, then start comparing it with other addresses